### PR TITLE
Adding gallery view for example notebooks

### LINKED ISSUE #467: Show notebooks in documentation from table of content and gallery
From #461, notebooks are included in the documentation using simple hyperlinks from the `examples.ipynb` notebook and show as html pages. However, `nbsphinx` allows to use both [table of contents](https://nbsphinx.readthedocs.io/en/0.8.0/subdir/toctree.html), RST-like links and the use of thumbnail [galleries](https://nbsphinx.readthedocs.io/en/0.8.0/subdir/gallery.html) similarly to sphinx-gallery. Implementing both would be great. The catch with respect to TOC links, is that they should be linkable also if someone is opening the `examples.ipynb` notebook, and not only looking at `examples.html` from the documentation (on Read The Docs or built locally). Probably keeping both the hyperlinks and rst-like links could work fine.

### PR DESCRIPTION
**Pull request by crazy4pi314**
Description
-----------
Fixes #467
 
Looks like this now: 
image

Need to make sure other images are supplied for notebooks that dont generate plots/graphics.

Checklist
-----------

Check off the following once complete (or if not applicable) after opening the PR. The PR will be reviewed once this checklist is complete and all tests are passing.

- [x] I added unit tests for new code.
- [x] I used [type hints](https://www.python.org/dev/peps/pep-0484/) in function signatures.
- [x] I used [Google-style](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html) docstrings for functions.
- [x] I [updated the documentation](../blob/master/docs/CONTRIBUTING_DOCS.md) where relevant.

If some items remain, you can mark this a [draft pull request](https://github.blog/2019-02-14-introducing-draft-pull-requests/).

### CODE REVIEW

**Comment by crazy4pi314** [comment]
The next thing here is to add custom thumbnails for some of the notebooks that dont generate images: https://sphinx-gallery.github.io/stable/configuration.html#using-a-custom-default-thumbnail

**Comment by crazy4pi314** [comment]
@rmlarose @nathanshammah @andreamari Is this something we want in 0.10 or should I bump it?

**Comment by rmlarose** [comment]
Would be great to have asap imo!

**Comment by andreamari** [comment]
It would be great to have this, I like it!
However I don't think it is critical for 0.10. It can happen later if more time is needed. 

@crazy4pi314  Pennylane has a similar gallery.  I remember that we used to manually add thumbnails as png images, independently of what is generated by the notebook.  Sometimes they were just copied from the plots of notebook, sometimes modified versions of plots (e.g. cropped parts), sometimes they were simply unrelated images. In any case they were not self-generated.  This is just FYI, it doesn't mean that we must follow the same approach.

**Comment by crazy4pi314** [comment]
What it looks like now, set thumbnails manually for now, can work on a future upgrade to get them automatically scraping.

image

File: docs/source/examples/examples.myst [hunk 1]
```diff
@@ -1,15 +1,14 @@
...
+.. nbgallery::
+
+    ibmq-backends.rst
+    hamiltonians.rst
+    simple_landscape.rst
+    maxcut-demo.myst
+    pyquil_demo.myst
+    cdr_api.md
+    mitiq-paper/mitiq-paper-codeblocks
+    <!-- pec-tutorial.myst -->
```
Comment (andreamari) [inline docs/source/examples/examples.myst:13]:
```suggestion
pec-tutorial.myst
```
You probably did it to speedup the docs build :-)

File: docs/source/conf.py [hunk 2]
```diff
@@ -316,3 +318,21 @@ def get_incollection_template(self, e):
...
+    '.mystnb': ['jupytext.reads', {'fmt': 'mystnb'}],
+}
+nbsphinx_execute = 'always'
+
+nbsphinx_thumbnails = {
+    'examples/ibmq-backends': '_images/unitary_fund_logo.png',
+    'examples/hamiltonians': '_images/vqe-cirq-pauli-sum-mitigation-plot.png',
+    'examples/simple_landscape': '_images/simple_landscape_13_0.png',
+    'examples/maxcut-demo': '_images/maxcut-demo_48_1.png',
+    'examples/pyquil_demo': '_images/pyquil_demo_11_0.png',
```
Comment (andreamari) [inline docs/source/conf.py:334]:
```suggestion
    'examples/pyquil_demo': '_images/unitary_fund_logo.png',
```
At the moment,  the pyquil example is not executed in readthedocs. This means that also the image is not generated (see here https://mitiq.readthedocs.io/en/crazy4pi314-467/examples/examples.html ).    Until we figure out how to run Rigetti QVMs in readthedocs, it is probably better to use default image.

**Review by andreamari — COMMENTED** [review]
Thanks @crazy4pi314.   
Apart from the two comments that I left below,  this LGTM!

Question that may be considered in this PR or elsewhere: Do we want to replace the unitary fund logo with the mitiq logo that we already have in the README ?  I think the Mitiq logo is more appropriate for the thumbnails and for the top-left corner of the docs too.

**Comment by crazy4pi314** [comment]
> Do we want to replace the unitary fund logo with the mitiq logo that we already have in the README

I would like to, and almost did it :) Let me do that and we can finish this one.

**Pull request closed without merging**